apps/web/src/pages/templates/filter/FilterModal.tsx

```diff
@@ -92,81 +100,45 @@ export function FilterModal({
-                }}
-              />
-            </Grid.Col>
-            <Grid.Col span={2}>
-              <Controller
-                control={control}
-                name={`steps.${stepIndex}.filters.0.children.${index}.operator`}
-                render={({ field }) => {
-                  return (
-                    <Select
-                      placeholder="Operator"
-                      data={[
-                        { value: 'EQUAL', label: 'Equal' },
-                        { value: 'NOT_EQUAL', label: 'Not equal' },
-                        { value: 'LARGER', label: 'Larger' },
-                        { value: 'SMALLER', label: 'Smaller' },
-                        { value: 'LARGER_EQUAL', label: 'Larger or equal' },
-                        { value: 'SMALLER_EQUAL', label: 'Smaller or equal' },
-                        { value: 'IN', label: 'Contains' },
-                        { value: 'NOT_IN', label: 'Not contains' },
-                      ]}
-                      {...field}
-                    />
-                  );
-                }}
-              />
-            </Grid.Col>
-            <Grid.Col span={2}>
-              <Controller
-                control={control}
-                name={`steps.${stepIndex}.filters.0.children.${index}.value`}
-                render={({ field, fieldState }) => {
-                  return <Input {...field} error={fieldState.error?.message} placeholder="Value" />;
-                }}
-              />
-            </Grid.Col>
-            <Grid.Col span={1}>
-              <DeleteStepButton
-                variant="outline"
-                size="md"
-                mt={30}
-                onClick={() => {
-                  remove(index);
-                }}
-              >
-                <Trash />
-              </DeleteStepButton>
-            </Grid.Col>
-          </Grid>
+          <div key={index}>
+            <Grid columns={10} key={item.id} align="center" gutter="xs">
+              <Grid.Col span={3}>
+                <Controller
+                  control={control}
+                  name={`steps.${stepIndex}.filters.0.children.${index}.on`}
+                  render={({ field }) => {
+                    return (
+                      <Select
+                        placeholder="On"
+                        data={[
+                          { value: 'payload', label: 'Payload' },
+                          { value: 'subscriber', label: 'Subscriber' },
+                          { value: 'webhook', label: 'Webhook' },
+                        ]}
+                        {...field}
+                        onChange={handleOnChildOnChange(index)}
+                      />
+                    );
+                  }}
+                />
+              </Grid.Col>
+              {filterFieldOn === 'webhook'
+                ? getWebhookUrlGrid(control, stepIndex, index)
+                : getEqualityGrid(filterFieldOn, control, stepIndex, index, remove)}
```

Comment:
And here you would just use the component:

```suggestion
              {filterFieldOn === 'webhook'
                ? <WebhookForm control={control} stepIndex={stepIndex} index={index} />
                : <Equality Grid (filterFieldOn, control, stepIndex, index, remove) />
```